specs/oracle.md

```diff
@@ -0,0 +1,188 @@
+is then gossiped to other validators.
+
+When a node receives a vote extension from a peer, cometbft will call `verify_vote_extension`.
+The app performs basic validation on the vote extension, ensuring it doesn't exceed
+the maximum byte size, that it's a valid `OracleVoteExtension`, and the number of
+prices doesn't exceed the maximum. If the VE passes this check, it's added to the
+node's local VE view for that round.
+
+#### `prepare_proposal`
+
+The proposer of a block receives all the vote extensions in its local view from
+the previous round inside [`prepare_proposal`](https://github.com/astriaorg/astria/blob/b2083b4a82195dc9be1e85f31cea14c724b8b4ec/crates/astria-sequencer/src/app/vote_extension.rs#L210),
+specifically the `local_last_commit`
+field which contains the previous round's commit (>2/3 voting power set of votes
+from validators) as well as their vote extensions. The proposer then checks if >2/3
+voting power submitted valid oracle data vote extensions. If not, the proposer
+omits the `extended_commit_info` (proposed canonical extended vote set) from the
+block. Otherwise, it includes it in the proposed block at a specific block data index.
+
+Note that in the case where not enough validators partake in the oracle protocol,
+liveness of the chain is unaffected. Blocks continue to be produced, but oracle
+prices are not updated.
+
+#### `process_proposal`
+
+Other validators receive the proposed block in `process_proposal`, which calls
+[`validate_proposal`](https://github.com/astriaorg/astria/blob/b2083b4a82195dc9be1e85f31cea14c724b8b4ec/crates/astria-sequencer/src/app/vote_extension.rs#L271).
+If it contains
+non-empty extended commit info, it validates it by checking that the signature
+for each vote extension is valid, and corresponds to an actual validator. It checks
+that each VE is itself a valid `OracleVoteExtension`. It also checks that >2/3
+voting power submitted valid VEs. If these checks pass, then the validator
+accepts the proposal (assuming the rest of the block is also valid), otherwise,
+it rejects the proposal. If the block's extended commit info is empty, then we
+assume the proposer did not receive the necessarily amount of VEs to create a
+valid VE set, and skip oracle validation to ensure liveness.
+
+#### `finalize_block`
+
+When a block with an extended commit info set is [finalized](https://github.com/astriaorg/astria/blob/b2083b4a82195dc9be1e85f31cea14c724b8b4ec/crates/astria-sequencer/src/app/mod.rs#L1093),
+the node uses the price
+data inside the vote extensions to [calculate updated prices](https://github.com/astriaorg/astria/blob/b2083b4a82195dc9be1e85f31cea14c724b8b4ec/crates/astria-sequencer/src/app/vote_extension.rs#L577)
+for each currency pair.
+For each currency pair, the node takes the validator-power-weighted median of
+the prices to calculate the new price, which is then stored in the app state.
+
+### Conductor
```

Comment:
Is it in any form a requirement that the price feed is prepended to the rest of the rollup transactions?

It's how conductor is implemented, but it's not mentioned here.